From 26a7eef5a6836e7c1e5d0887a89f45ed645d6f5c Mon Sep 17 00:00:00 2001
From: bennibbelink <79653949+bennibbelink@users.noreply.github.com>
Date: Wed, 2 Oct 2024 07:56:09 -0400
Subject: [PATCH] create .so symlink to libraries instead of second
 installation; hardcode symlinks as .so so python can find them on macos

---
 src/CMakeLists.txt | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a41e68e..5c9eb7e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -162,9 +162,15 @@ if(Cython_FOUND)
           TARGETS ${name}
           LIBRARY DESTINATION lib
           )
-        INSTALL(
-          TARGETS ${name}
-          LIBRARY DESTINATION ${PYTHON_SITE_PACKAGES}
+        # Add symlinks to PYTHON_SITE_PACKAGES so python can find our cython modules
+        # Python will only import libraries with .so extensions
+        # so we hardcode the symlink with .so even on APPLE
+        INSTALL(CODE "
+            EXECUTE_PROCESS(
+                COMMAND ${CMAKE_COMMAND} -E create_symlink
+                ${CMAKE_INSTALL_PREFIX}/lib/${name}${suffix}
+                ${PYTHON_SITE_PACKAGES}/${name}.so
+            )"
           )
 
         copy_when_diff(${_generated_h} ${_h_file})
-- 
2.56.0

